### DISCUSSION TITLE
Initial notes

### DISCUSSION THREAD

**Discussion by greenc-FNAL** [FNALssi/spack-subspack#1] [General]
1. Given other connotations in wide colloquial use, there are probably better choices for an abbreviated package shortcut than, "sext."
2. In what way(s) could the desired functionality be provided by (or leverage, or learn from) `spack  clone`?
3. Suggest using Python equivalent to `spack env list` to find environments, since they can be trivially created out-of-tree.

**Comment by marcmengel** [comment]
initial responses :-) 
1. Good point, fixed.
2. `spack clone ` simply does some error checks for existing files and then runs a few git commands trying to setup with the same (or given) upstream.    I suppose it's more spack-ish to use  
```
        git = spack.util.git.git(required=True)
        git("init", "--shared", "-q")
```
than to os.system("git ...") so  perhaps I'll do that. 
3. If you do `spack env create --dir, ` spack env list doesn't list it, nor can you spack activate it by name,  you can only find it by path.  Such a directory environment should I believe work from the subspack as well.
